spec/graphql/schema/introspection_system_spec.rb

```diff
@@ -52,5 +52,14 @@
       ensembles_field = query_type["fields"].find { |f| f["name"] == "ensembles" }
       assert_equal [], ensembles_field["args"]
     end
+
+    it "runs the introspection query and the result contains a edge field that has non-nullable node" do
+      res = NonNullableDummy::Schema.execute(GraphQL::Introspection::INTROSPECTION_QUERY)
+      assert res
+      edge_type = res["data"]["__schema"]["types"].find { |t| t["name"] == "NonNullableNodeEdge" }
+      node_field = edge_type["fields"].find { |f| f["name"] == "node" }
+      assert_equal "NON_NULL", node_field["type"]["kind"]
+      assert_equal "NonNullableNode", node_field["type"]["ofType"]["name"]
+    end
```

Comment:
I thought writing tests here would not be the best, but I did this because it is unprecedented.